src/Bicep.Core/Parsing/ParamsParser.cs

```diff
@@ -100,14 +103,22 @@ private ExtendsDeclarationSyntax ExtendsDeclaration()
             return new ExtendsDeclarationSyntax(keyword, path);
         }
 
-        private SyntaxBase ParameterAssignment()
+        private SyntaxBase ParameterAssignment(IEnumerable<SyntaxBase> leadingNodes)
         {
             var keyword = ExpectKeyword(LanguageConstants.ParameterKeyword);
             var name = this.IdentifierWithRecovery(b => b.ExpectedParameterIdentifier(), RecoveryFlags.None, TokenType.Identifier, TokenType.NewLine);
-            var assignment = this.WithRecovery(this.Assignment, GetSuppressionFlag(name), TokenType.NewLine);
-            var value = this.WithRecovery(() => this.Expression(ExpressionFlags.AllowComplexLiterals), GetSuppressionFlag(assignment), TokenType.NewLine);
+            var assignment = this.reader.Peek().Type switch
+            {
+                TokenType.NewLine => null,
+                _ => this.WithRecovery(this.Assignment, GetSuppressionFlag(name), TokenType.NewLine),
+            };
+            var value = this.reader.Peek().Type switch
+            {
+                TokenType.NewLine => null,
```

Comment:
If there's a newline after the =, shouldn't we emit a parsing diagnostic? Setting it to null here means that syntax like `param foo =\n` won't trigger any errors.

Reply:
`throws Error BCP338: Failed to evaluate parameter "foo": Parameter value is null` is a semantic error, but we should emit a syntax error here to block invalid syntax `param foo =\n`.

Reply:
At syntax level, we should only allow the following cases:
- `param foo = <expression>`
- `param foo`